gravitee-io/gravitee-access-management
# AM-90: mfa verify brute force detection implemented

### jira issue: https://graviteecommunity.atlassian.net/browse/AM-90


### how to test the feature:

1. Create email or sms factor.
2. Enable the factor in an application and enroll, then verify for a user to make sure everything is fine as before.
3. Now from `User Accounts` menu enable brute force detection from the newly added `MFA` section. 
4. For testing purpose input could be max challenge attempt 3, reset time 1 minutes and enable send email option
5. Save the settings.
6. Now trigger login flow and during MFA verify steps enter 3 wrong value and try to verify. You should receive `invalid code` error as before, no behaviour change in this step.
7. Provide wrong/right code value on the 4th time and you should receive `maximum attempts reached` error.
8. An email will be sent to the user account  along with an audit log entry for the attempt reached entry.
9. Try again after 1 minute with correct value and verification should be completed successfully.
